pkg/webhook/controlplane/ensurer.go

```diff
@@ -39,4 +45,51 @@ func (e *ensurer) InjectClient(client client.Client) error {
 	return nil
 }
 
-// TODO: modify control-plane arguments here, like api-server args etc.
+// EnsureKubeAPIServerDeployment ensures that the kube-apiserver deployment conforms to the provider requirements.
+func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, ectx genericmutator.EnsurerContext, new, _ *appsv1.Deployment) error {
+	template := &new.Spec.Template
+	ps := &template.Spec
+
+	if c := extensionswebhook.ContainerWithName(ps.Containers, "kube-apiserver"); c != nil {
+		ensureKubeAPIServerCommandLineArgs(c)
+	}
+	return nil
+}
+
+// EnsureKubeControllerManagerDeployment ensures that the kube-controller-manager deployment conforms to the provider requirements.
+func (e *ensurer) EnsureKubeControllerManagerDeployment(ctx context.Context, _ genericmutator.EnsurerContext, new, _ *appsv1.Deployment) error {
+	template := &new.Spec.Template
+	ps := &template.Spec
+
+	if c := extensionswebhook.ContainerWithName(ps.Containers, "kube-controller-manager"); c != nil {
+		ensureKubeControllerManagerCommandLineArgs(c)
+	}
+	return nil
+}
+
+func ensureKubeAPIServerCommandLineArgs(c *corev1.Container) {
+	c.Command = extensionswebhook.EnsureNoStringWithPrefix(c.Command, "--cloud-provider=")
+}
+
+func ensureKubeControllerManagerCommandLineArgs(c *corev1.Container) {
+	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--cloud-provider=", "external")
+}
+
+// EnsureKubeletServiceUnitOptions ensures that the kubelet.service unit options conform to the provider requirements.
+func (e *ensurer) EnsureKubeletServiceUnitOptions(ctx context.Context, _ genericmutator.EnsurerContext, new, _ []*unit.UnitOption) ([]*unit.UnitOption, error) {
+	if opt := extensionswebhook.UnitOptionWithSectionAndName(new, "Service", "ExecStart"); opt != nil {
+		command := extensionswebhook.DeserializeCommandLine(opt.Value)
+		command, err := e.ensureKubeletCommandLineArgs(ctx, command)
+		if err != nil {
+			return nil, err
+		}
+		opt.Value = extensionswebhook.SerializeCommandLine(command, 1, " \\\n    ")
+	}
+	return new, nil
+}
+
+func (e *ensurer) ensureKubeletCommandLineArgs(ctx context.Context, command []string) ([]string, error) {
```

Comment:
Returning an error here is useless. Please change the signature to be:
```func (e *ensurer) ensureKubeletCommandLineArgs(ctx context.Context, command []string) []string```

Reply:
You are right, will do.

Reply:
Done